garage/tf/models/gaussian_mlp_model.py

```diff
@@ -0,0 +1,194 @@
+                mean_var = mean_std_model.output[..., self._output_dim:]
+                std_param_var = mean_std_model.output[..., :self._output_dim]
+            else:
+                mean_model = mlp2(
+                    input_var=input_var,
+                    output_dim=self._output_dim,
+                    hidden_sizes=self._hidden_sizes,
+                    hidden_nonlinearity=self._hidden_nonlinearity,
+                    output_nonlinearity=self._output_nonlinearity,
+                    scope="mean_network")
+                mean_var = mean_model.output
+
+                if self._adaptive_std:
+                    std_model = mlp2(
+                        input_var=input_var,
+                        output_dim=self._output_dim,
+                        hidden_sizes=self._std_hidden_sizes,
+                        hidden_nonlinearity=self._std_hidden_nonlinearity,
+                        output_nonlinearity=self._output_nonlinearity,
+                        output_b_init=b,
+                        scope="std_network")
+
+                    std_param_var = std_model.output
+                else:
+                    p = tf.constant_initializer(self._init_std_param)
+                    std_param_var = ParameterLayer(
+                        length=self._output_dim,
+                        initializer=p,
+                        trainable=self._learn_std,
+                        scope="std_network")(input_var)
+
+            with tf.variable_scope("std_limits"):
+                if self._min_std_param:
+                    std_param_var = Lambda(
+                        lambda x, f, p: f(x, p),
+                        arguments={
+                            'f': tf.maximum,
+                            'p': self._min_std_param
+                        })(std_param_var)
+                if self._max_std_param:
+                    std_param_var = Lambda(
+                        lambda x, f, p: f(x, p),
+                        arguments={
+                            'f': tf.minimum,
+                            'p': self._max_std_param
+                        })(std_param_var)
+
+        with tf.variable_scope("std_parameterization"):
+            if self._std_parameterization == "exp":
+                std_var = Lambda(
+                    lambda x, f: f(x), arguments={'f': tf.exp})(std_param_var)
+            elif self._std_parameterization == "softplus":
+                std_var = Lambda(
+                    lambda x, f, g: f(1 + g(x)),
+                    arguments={
+                        'f': tf.log,
+                        'g': tf.exp
+                    })(std_param_var)
+            else:
+                raise NotImplementedError
+
+        sample_var = Lambda(
+            lambda x, f, s: f(x[0], x[1]).sample(seed=s),
+            arguments={
+                'f': self._dist,
+                's': ext.get_seed()
+            })([mean_var, std_param_var])
```

Comment:
Could you pass a `name=MultivariateNormalDiagDistribution` to Lambda here?

Reply:
Yes.

Reply:
Could you show what a tf graph would like if you don't pass a name to the layer? It would be better to pass a name here otherwise the name in the config would be just Lambda. It is more informative to show it is a distribution layer